doc/nrf/create_application.rst

```diff
@@ -0,0 +1,224 @@
+.. _create_application:
+
+Create an application
+#####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The process for creating an application in the |NCS| follows Zephyr's :ref:`Application Development principles <zephyr:application>` and the |NCS|'s :ref:`dev-model`.
+The following sections provide basic information based on these two sources.
+
+.. _create_application_structure:
+
+Application structure
+*********************
+
+Here are the files that most of the Zephyr and |NCS| applications include in its :file:`<app>` directory:
+
+.. ncs-include:: develop/application/index.rst
+   :docset: zephyr
+   :start-after: Here are the files in a simple Zephyr application:
+   :end-before: These contents are:
+
+You can read more about each of these files in Zephyr's :ref:`Application Development overview <zephyr:application>`.
+
+.. _create_application_types:
+
+Application types
+*****************
+
+Based on where the :file:`<app>` directory is located, you will usually work with one of two types of applications in the |NCS|: workspace or freestanding.
+These types are adopted from :ref:`Zephyr application types <zephyr:zephyr-app-types>`.
+
+.. _create_application_types_workspace:
+
+Workspace application
+=====================
+
+A workspace application is located inside a west workspace, but outside of the repositories of the SDK.
+This type of application uses its own copy of the |NCS|.
+It specifies the |NCS| version through the :file:`west.yml` `west manifest file`_, which is located in the application :file:`<app>` directory.
+
+The workspace application has the following structure:
+
+.. code-block:: none
+
+   <home>/
+   └─── ncs/
+      ├─── .west/
+      ├─── nrf/
+      ├─── zephyr/
+      ├─── ...
+      └─── <app>/
+         ├── src/
+         ├── ...
+         └── west.yml
+
+You want to use this application type for the following development cases:
+
+* You want to take advantage of west to manage your own set of repositories.
+* You want to make changes to one or more of the repositories of the |NCS| when working on the application.
+* You want to develop a project that involves more than one build target, for example using a mesh networking protocol like :ref:`ug_matter` or :ref:`ug_bt_mesh`.
```

Comment:
I'd also add that this model allows to develop most features without patching the NCS tree, e.g. oot boards, drivers, socs, etc. So in practice, any "big" project will need this model, except they prefer to keep a patched NCS.